src/coreclr/jit/lsrabuild.cpp

```diff
@@ -3879,7 +3882,7 @@ int LinearScan::BuildBinaryUses(GenTreeOp* node, SingleTypeRegSet candidates)
     {
 
 #ifdef TARGET_XARCH
-        if (op2->isContainedIndir() && varTypeUsesFloatReg(op1) && candidates == RBM_NONE)
+        if (op2->isContainedIndir() && candidates == RBM_NONE && !canUseApxRegs)
```

Comment:
```suggestion
        if (op2->isContainedIndir() && (candidates == RBM_NONE) && !canUseApxRegs)
```

Reply:
It seems here, we should have `if ((op2->isContainedIndir() && (candidates == RBM_NONE)) || !canUseApxRegs)`...basically even if `op2` is not contained indir and `candidates != RBM_NONE`, we should still use `lowGprRegs` if we cannot use APX registers.